src/driver/virtualbox.rs

```diff
@@ -3,103 +3,216 @@ use std::error::Error;
 use fdp::{RegisterType, FDP};
 
 use crate::api::{
-    DriverInitParam, Introspectable, Registers, SegmentReg, SystemTableReg, X86Registers,
+    DriverError, DriverInitParam, Introspectable, Registers, SegmentReg, SystemTableReg,
+    X86Registers,
 };
 
+#[derive(thiserror::Error, Debug)]
+pub enum VirtualBoxDriverError {
+    #[error(transparent)]
+    OtherError(#[from] Box<dyn Error>),
```

Comment:
My idea was to first get the error handling in our dependencies right, so we don't have to do things like this, for example. (Also, it's a lot less controversial than error handling in this crate).